### PR TITLE
Converting array data to json within the create method of paymentlink…

### PR DESCRIPTION
**Pull request by neera11**
Converting array data to json within the create method of paymentlink and passing to request method

### CODE REVIEW

File: src/Payment_link.php [hunk 1]
```diff
@@ -16,6 +16,8 @@ class Payment_link extends Entity
      */
     public function create($attributes = array())
     {
```
Comment (ramth05) [inline src/Payment_link.php:18]: @neera11 , please update the class name

File: src/Payment_link.php [hunk 2]
```diff
@@ -16,6 +16,8 @@ class Payment_link extends Entity
...
+        $attributes = json_encode($attributes);
```
Comment (ramth05) [inline src/Payment_link.php:19]: @abhayp @neera11 , why we need this conversion when other API on RZP support for both json and form data as well.

File: src/Payment_link.php [hunk 3]
```diff
@@ -16,6 +16,8 @@ class Payment_link extends Entity
...
+        
         $additionHeader['Content-Type'] = 'application/json';
```
Comment (ramth05) [inline src/Payment_link.php:21]: @neera11 , Request class already have way to add additional headers. why this new function required.

File: src/Payment_link.php (hunk 1 above)
Comment (neera11) [reply src/Payment_link.php:18]: updated

File: src/PaymentLink.php [hunk 4]
```diff
@@ -5,17 +5,19 @@
...
      * @param array $attributes
      *
-     * @return Payment_link
+     * @return PaymentLink
      */
     public function create($attributes = array())
     {
+        $attributes = json_encode($attributes);
+        
         $additionHeader['Content-Type'] = 'application/json';
```
Comment (ramth05) [inline src/PaymentLink.php:21]: @neera11 ,  can we use Request `addHeader` method here.

File: src/Payment_link.php (hunk 2 above)
Comment (neera11) [reply src/Payment_link.php:19]: @abhayp Can you please clarify.As discussed earlier it is constraint due to new APIs written in go lang.

File: src/Payment_link.php (hunk 3 above)
Comment (neera11) [reply src/Payment_link.php:21]: @ramth05 
I corrected this, now using Request call addHeader function to add additional header

File: src/Entity.php [hunk 5]
```diff
@@ -88,11 +85,11 @@ protected function snakeCase($input)
      *
      * @return Entity
      */
-    protected function request($method, $relativeUrl, $data = null, $additionHeader = array())
+    protected function request($method, $relativeUrl, $data = null)
     {
         $request = new Request();
 
-        $response = $request->request($method, $relativeUrl, $data, $additionHeader);
```
Comment (ramth05) [inline src/Entity.php:95]: @neera11 , Please make sure you revert all the changes related to this from last PR merged.

File: src/PaymentLink.php [hunk 6]
```diff
@@ -5,20 +5,22 @@
...
      */
     public function create($attributes = array())
     {
-        $additionHeader['Content-Type'] = 'application/json';
+        $attributes = json_encode($attributes);
 
-        parent::setAdditionHeader($additionHeader) ;
+        $request = new Request;
+        
+        $request->addHeader('Content-Type', 'application/json');
```
Comment (ramth05) [inline src/PaymentLink.php:23]: addHeader is static method can be used directly `Request::addHeader('Content-Type', 'application/json');`

Comment (neera11) [reply src/PaymentLink.php:23]: Yes we can, corrected

File: src/Entity.php (hunk 5 above)
Comment (neera11) [reply src/Entity.php:95]: yes i reverted all changes

**Review by ramth05 — APPROVED** [review]
LGTM, Please test it with composer.

**Review by ramth05 — APPROVED** [review]
LGTM, please test it throughly and with composer as well.

**Pull request merged**